cmd/entrypoint/cmd/root.go

```diff
@@ -189,6 +187,41 @@ func removedEnv() []string {
 	return removedEnv
 }
 
+func runService(ctx context.Context) error {
+	for {
+		code, err := runCommand(ctx, runnerDir, listenerCommand, "run", "--startuptype", "service", "--once")
+		if _, ok := err.(*exec.ExitError); !ok {
+			return err
+		}
+
+		// This logic is based on the following code.
+		// ref: https://github.com/actions/runner/blob/v2.278.0/src/Misc/layoutbin/RunnerService.js
+		fmt.Println("Runner listener exited with error code", code)
+		switch code {
+		case 0:
+			fmt.Println("Runner listener exit with 0 return code, stop the service, no retry needed.")
+			metrics.IncrementListenerExitState(metrics.Successful)
+			return nil
+		case 1:
+			fmt.Println("Runner listener exit with terminated error, stop the service, no retry needed.")
+			metrics.IncrementListenerExitState(metrics.TerminatedError)
```

Comment:
If the runner exits with the Terminated Error, the process is immediately down and this changed metrics value here would not be corrected I think.

Reply:
Removed TerminatedError and Successful metrics that are not going to be used.